### PR TITLE
Basic support of the Eclipse Ceylon language

### PR DESCRIPTION
**Pull request by davidfestal**
### What does this PR do?

This PR adds basic support of the Eclipse Ceylon language in Eclipse Che:
- language description,
- syntax highlighting,
- project creation.

### What issues does this PR fix or reference?

It refers no issue and is a new feature contribution

#### Release Notes

Basic support of the Eclipse Ceylon language in Eclipse Che with:
- language description,
- syntax highlighting,
- project creation.

### CODE REVIEW

**Comment by skabashnyuk** [comment]
ci-test

**Comment by skabashnyuk** [comment]
@davidfestal please let us know when this PR would be ready to review.

**Comment by davidfestal** [comment]
ci-test

**Comment by davidfestal** [comment]
ci-test

**Comment by davidfestal** [comment]
> please let us know when this PR would be ready to review.

@skabashnyuk according to the [last build](https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/385/), it seems ready to review.

File: plugins/plugin-ceylon/che-plugin-ceylon-lang-ide/pom.xml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                <configuration>
+                    <moduleName>org.eclipse.che.plugin.ceylon.Ceylon</moduleName>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>com.mycila</groupId>
+                <artifactId>license-maven-plugin</artifactId>
+                <configuration>
+                    <excludes>
+                        <exclude>**/ceylon.svg</exclude>
```
[the remark this answers is not in the document]
Comment (davidfestal) [reply plugins/plugin-ceylon/che-plugin-ceylon-lang-ide/pom.xml:91]: It's the extact same file as the one already added in this PR: https://github.com/eclipse/che/pull/5844#issuecomment-320647755

Also see this answer to a question from @benoitf : https://github.com/eclipse/che/pull/5844#issuecomment-320612516

File: plugins/plugin-ceylon/che-plugin-ceylon-lang-server/pom.xml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>com.mycila</groupId>
+                <artifactId>license-maven-plugin</artifactId>
+                <configuration>
+                    <excludes>
+                        <!-- The files don't need a license since they will
+                        be added as templates inside the user projects -->
+                        <exclude>src/main/resources/project/**</exclude>
```
[the remark this answers is not in the document]
Comment (benoitf) [reply plugins/plugin-ceylon/che-plugin-ceylon-lang-server/pom.xml:73]: IMHO generated code is unlicensed. Each user is making the choice of the license to use for their project.
Generator library is licensed but not why user find as its "project files"

**Review by skabashnyuk — APPROVED** [review]

File: plugins/plugin-ceylon/pom.xml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    Contributors:
+      Red Hat, Inc. - initial API and implementation
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>che-plugin-parent</artifactId>
+        <groupId>org.eclipse.che.plugin</groupId>
+        <version>6.7.0-SNAPSHOT</version>
```
Comment (vparfonov) [inline plugins/plugin-ceylon/pom.xml]: here need to set new version 6.8.0-SNAPSHOT

**Review by vparfonov — APPROVED** [review]

**Comment by davidfestal** [comment]
ci-test

**Comment by davidfestal** [comment]
ci-test

**Review by benoitf — APPROVED** [review]

**Review by riuvshin — APPROVED** [review]

**Pull request merged**